### PR TITLE
UID-79 Add "handler surface" page

### PR DESCRIPTION
**Pull request by MikeTaylor**
**DO NOT MERGE**

This "works", but only with whatever handler-event is named as the initial state in [the `useState` call at the top of the `HandlerSurface` functional component](https://github.com/folio-org/ui-developer/blob/b1b49fca501734a86a7c5d434b26011310025dc5/src/settings/HandlerSurface.js#L8). When you change the event name interactively, the rendering does not change with it — unlike the analogous situation with the "plugin surface" page.

I think this is because [the `<HandlerManager>` component provided by stripes-core](https://github.com/folio-org/stripes-core/blob/master/src/components/HandlerManager/HandlerManager.js) consults the `event` parameter only in the constructor, deriving the set of event-handlers at construction time and not changing them on subsequent renders.

Is there an easy way that I can (without changing `<HandlerManager>`) force this new page to re-instantiate its `<HandlerManager>` component from fresh every time the event-name changes? Or is that just a confused question?

### CODE REVIEW

**Comment by MikeTaylor** [comment]
This now works, thanks to passing a `key={handlerEvent}` prop to `<HandlerManager>` to force a remount when the event name changes. @zburke Please review so I can merge.

**Comment by MikeTaylor** [comment]
Wait — has SonarCloud's BDSM insistence on "quality" now become a _required_ condition for merging in ui-developer? When did this happen? On whose authority? And how can we back out this moronic policy?

File: src/settings/index.js [hunk 1]
```diff
@@ -102,6 +103,12 @@ const pages = [
     component: PluginSurface,
     perm: 'ui-developer.settings.plugin-surface',
   },
+  {
+    route: 'handler-surface',
+    labelId: 'ui-developer.handler-surface',
+    component: HandlerSurface,
+    // perm: 'ui-developer.settings.handler-surface',
```
Comment (zburke) [inline src/settings/index.js:110]: Please enable this before you merge.

**Review by zburke — APPROVED** [review]
This looks helpful; thank you! Please enable the permission before merging.

File: src/settings/index.js (hunk 1 above)
Comment (MikeTaylor) [reply src/settings/index.js:110]: If I enable the permission, I won't be able to use the feature.

**Comment by MikeTaylor** [comment]
Oh well: I enabled the check for this merge, and will just re-disable it in my local copy until the platform I'm using as my server gets rebuilt.

**Pull request merged**